### DISCUSSION TITLE
Issue with Record Segment Duration Configuration

### DISCUSSION THREAD

**Discussion by CheSauRie** [bluenviron/mediamtx#3901] [General]
Hi,

I am currently streaming an .mp4 file that is 10 minutes long, and I have configured recordSegmentDuration=0h2m0s in my setup. However, when I check the recorded files on the playback server, I receive a single 10-minute recording instead of multiple 2-minute segments as expected.

Could you please explain why this is happening and if there is a way to ensure the recordings are segmented into 2-minute chunks?

Thank you for your help!

**Comment by aler9** [comment]
Hello, recording segments are ideally splitted by using the duration set inside `recordSegmentDuration`, but in reality they have to start with an I-frame, which is a special frame from which the stream can be decoded independently from past frames.

The server waits that `recordSegmentDuration` has elapsed and then waits for an I-frame to appear, before closing current segment and opening a new one.

Therefore, the actual duration of a record segment is influenced by the "I-frame interval", which is a setting that is present in most cameras and is also present in any video encoder. Make sure that your MP4 has I-frames every minute or re-encode it.
